worker/firewaller/firewaller.go

```diff
@@ -1011,23 +1049,63 @@ func (fw *Firewaller) flushGlobalPorts(rawOpen, rawClose firewall.IngressRules)
 	// Open and close the ports.
 	if len(toOpen) > 0 {
 		toOpen.Sort()
+		fw.logger.Infof("opening port ranges %v in environment", toOpen)
 		if err := fw.environFirewaller.OpenPorts(fw.cloudCallContextFunc(ctx), toOpen); err != nil {
 			// TODO(mue) Add local retry logic.
-			return err
+			return errors.Annotatef(err, "failed to open port ranges %v in environment", toOpen)
 		}
-		fw.logger.Infof("opened port ranges %v in environment", toOpen)
 	}
 	if len(toClose) > 0 {
 		toClose.Sort()
+		fw.logger.Infof("closing port ranges %v in environment", toClose)
 		if err := fw.environFirewaller.ClosePorts(fw.cloudCallContextFunc(ctx), toClose); err != nil {
 			// TODO(mue) Add local retry logic.
-			return err
+			return errors.Annotatef(err, "failed to close port ranges %v in environment", toOpen)
 		}
-		fw.logger.Infof("closed port ranges %v in environment", toClose)
 	}
 	return nil
 }
 
+func (fw *Firewaller) flushModel() error {
+	if fw.environModelFirewaller == nil {
```

Comment:
Don't need this check any more since it will not be nil if we call this

Reply:
There's still the possibility that in the future someone calls this and forgets this check. Having a descriptive error return early, whilst not strictly necessary, I think is worth keeping for the readability